tests/test_api/test_endpoints.py

```diff
@@ -987,6 +987,74 @@ def test_run_query_update_as_user(api_user_client):
             },
         )
 
+def test_run_query_aggregate_as_user(api_user_client):
+    """
+    Submit a request to aggregate data without the correct permissions. Then add the permissions
+    and submit the same request again, this time expecting a successful response.
+    """
+    mdb = get_mongo_db()
+    study_set = mdb.get_collection("study_set")
+
+    # Assert that the `study_set` collection does not already contain studies
+    # like the ones we're going to generate here. Then, generate 6 studies
+    # and insert them into the database.
+    #
+    # Note: The reason assertion is necessary is that some longstanding tests
+    #       in this repostory leave "residue" in the test database after they
+    #       run. See "FIXME" comments in this module for more details.
+    #
+    study_title = "My study"
+    nonexistent_study_title = "Nonexistent study"
+    assert study_set.count_documents({"title": study_title}) == 0
+    assert study_set.count_documents({"title": nonexistent_study_title}) == 0
```

Comment:
I think the three regular studies will be sufficient for this test, since the test is focused on the endpoint's permissions as opposed to how the endpoint handles existing versus nonexistent documents.

```suggestion
    study_title = "My study"
    assert study_set.count_documents({"title": study_title}) == 0
```